### ISSUE TITLE
Crash due to EqualizeDocumentLength() sometimes result in unequal lengths

### ISSUE THREAD

**Issue by bjornblissing** [Caphyon/clang-format-editor#29]
The function `DetectOperationPerLine()` tries to equalize the length of the incoming document strings if their length is unequal.
https://github.com/Caphyon/clang-format-editor/blob/f766b3f38f3c6cc4a526db558bb8e76ce1b10943/ClangFormatEditor/DiffStyle/DiffMatchPatchWrapper.cs#L190-L209

However, there is no check to make sure that resulting strings are _actually_ equal after the functions `EqualizeDocumentLength` have been run.

If the resulting list `outputLines` have more elements than then resulting list `inputLines` this will cause a crash inside the for-loop, since the `GetDiff()` function will try to index the `inputLines` variable outside outside the available list range.

My temporary fix was to skip elements outside the list range inside the for-loop:
```c#
      for (int index = 0; index < outputLines.Count; index++)
      {
        if (index >= inputLines.Count)
        {
            continue;
        }
        ...
```

However, the issue should probably be fixed inside the `EqualizeDocumentLength()` function.

**Comment by mariru27** [comment]
Hi @bjornblissing,

Sorry for delay answer. Thank you for reporting this issue
I can reproduce this problem, we are working on a fix
I will return to you when a release with this fix will be available

Have a nice day,
Marina

**Comment by mariru27** [comment]
Hi @bjornblissing,

A fix for this issue is available in Clang Format Editor v2023.4.2

Thank you for reporting this problem

Kind regards,
Marina
